projectdoc/implementation.tex

```diff
@@ -2,9 +2,104 @@
 \section{Implementation}
 \label{sec:Implementation}
 
-\subsubsection{Komponenten}
-\label{subsub:impl_Komponenten}
+\subsection{Komponenten}
+\label{impl:Komponenten}
 
+\subsection{QGIS Plugin}
+\label{impl:QGIS Plugin}
+TODO
+
+\subsection{PlazaRoute Container}
+\label{impl:PlazaRoute Container}
+TODO
+
+\subsubsection{Plaza Vorverarbeitung}
+\label{impl:Plaza Vorverarbeitung}
+TODO
+
+\subsubsection{Plaza Routing}
+\label{impl:Plaza Routing}
+In den folgenden Unterkapitel wird die Umsetzung der Plaza Routing Architektur, welche im Kapitel \ref{architektur:Plaza Routing} definiert ist, erläutert. 
+
+\paragraph{api}\label{impl:Plaza Routing api}~\\
+In diesem Kapitel wird auf die Umsetzung der api-Schicht in Abbildung \ref{fig:package_diagram_plaza_routing} Bezug genommen.
+
+TODO Flask Setup beschreiben
+
+\paragraph{Zusammenspiel und Verantwortlichkeit}\label{impl:Plaza Routing Zusammenspielund Verantwortlichkeit}~\\
+In Abbildung \ref{fig:sequence_diagram_plaza_routing_overview} sind die Interaktionen zwischen den Komponenten in einem Sequenz-Diagramm aufgeschlüsselt. Dabei wurde zur Wahrung der Übersichtlichkeit das Holen aller möglichen Routen in ein Unter-Squenz-Diagram in Abbildung \ref{fig:sequence_diagram_plaza_routing_route_combs} ausgelagert. Dabei geht es primär nicht um die Logik, die dahinter liegt, sondern um die Verantwortlichkeiten der Services. Wann sie ins Spiel kommen und für was sie benötigt werden, ist so gut ersichtlich.
+
+\begin{figure}[ht]
+    \centering
+    \includegraphics[width=0.7\linewidth]{projectdoc/img/sequence_diagram_plaza_routing_overview}
+    \caption[Plaza Routing Sequenz-Diagramm Übersich]{Plaza Routing Sequenz-Diagramm Übersicht}
+    \label{fig:sequence_diagram_plaza_routing_overview}
+\end{figure}
+
+\begin{figure}[ht]
+    \centering
+    \includegraphics[width=1\linewidth]{projectdoc/img/sequence_diagram_plaza_routing_route_comb}
+    \caption[Plaza Routing Sequenz-Diagramm Route Combinations]{Plaza Routing Sequenz-Diagramm Route Combinations}
+    \label{fig:sequence_diagram_plaza_routing_route_combs}
+\end{figure}
+
+So sieht man in Abbildung \ref{fig:sequence_diagram_plaza_routing_route_combs} gut, dass im Loop für eine Ausgangs-ÖV-Station die Verbindung von search.ch \cite{search_ch_route_api} geholt (siehe \ref{impl:Plaza Routing ÖV-Haltestellen eruieren}) und zwei Mal eine Fussgänger-Routing durchgeführt wird, jeweils vom aktuellen Standort zur ersten ÖV-Station und von der letzten ÖV-Station zur Destination. Diese Daten sind die Grundlagen für die Entscheidungsfindung der besten Route, welches in \ref{impl:Plaza Routing Beste Route eruieren} beschrieben ist.
```

Comment:
Dieses `\ref`s auf die Paragraphen unten funktionieren nicht richtig, da ein Paragraph keine Kapitelnummer hat. Im PDF wird dann einfach die Kapitelnummer `5.3.2` genommen, in dem sich alle diese Paragraphen befinden